File: 0801-0900/0868-Binary Gap/0868-Binary Gap.cpp
class Solution {
public:
    int binaryGap(int N) {
        int maxDist = 0, last = -1, curr = 0;
        while (N) {
            if (N & 1) {
                if (last != -1) {
                    maxDist = max(maxDist, curr - last);
                }
                last = curr;
            }
            ++curr;
            N >>= 1;
        }
        return maxDist;
    }
};

File: 0801-0900/0868-Binary Gap/0868-Binary Gap.py
class Solution:
    def binaryGap(self, N: int) -> int:
        maxDist = curr = 0
        prev = -1
        while N:
            if N & 1:
                if prev != -1:
                    maxDist = max(maxDist, curr - prev)
                prev = curr
            curr += 1
            N >>= 1
        return maxDist
